GoogleDataTransport/GDTCORLibrary/ClientMetrics/GDTCORDroppedEventsCounter.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import "GoogleDataTransport/GDTCORLibrary/ClientMetrics/GDTCOREventDropReason.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// Represents a number of events dropped for a particular mapping ID with a specified reason.
+/// The model is also used as a stored object for the persistent storage.
+@interface GDTCORDroppedEventsCounter : NSObject <NSSecureCoding>
+
+/// Number of events dropped.
+@property(nonatomic, readonly) NSUInteger eventCount;
+
+/// The reason why the events were dropped.
+@property(nonatomic, readonly) GDTCOREventDropReason dropReason;
```

Comment:
Do all of the events in the eventCount have the same dropReason?

Reply:
Yes, the intention is to maintain as many counters as many reasons and mapping IDs we have.